taichi/codegen/llvm/codegen_llvm.cpp

```diff
@@ -2218,7 +2232,7 @@ void TaskCodeGenLLVM::create_offload_struct_for(OffloadedStmt *stmt,
       // body tail: increment loop_index and jump to loop_test
       builder->SetInsertPoint(body_tail_bb);
 
-      if (spmd) {
+      if (spmd == OffloadSPMDType::nvgpu || spmd == OffloadSPMDType::amdgpu) {
```

Comment:
Based on the above mentioned refactor, you can remove `spmd` here and put the unified logic:
```
create_increment(loop_index, block_dim);
```